core/common-util/src/main/java/uk/gov/gchq/gaffer/commonutil/ByteArrayEscapeUtils.java

```diff
@@ -32,16 +35,28 @@ private ByteArrayEscapeUtils() {
     }
 
     /**
-     * Escapes the provided string so that it no longer contains the
+     * Escapes the provided byte[] so that it no longer contains the
      * Constants.DELIMITER character.
+     * After escaping the byte[] the appendAfterEscaping[] is appended unescaped.
      *
-     * @param bytes
-     *            the byte array to escape
+     * @param bytes               the byte array to escape
+     * @param appendAfterEscaping the bytes to append to the array after being escaped.
      * @return the escaped byte array
      */
-    public static byte[] escape(final byte[] bytes) {
-        final byte[] temp = new byte[2 * bytes.length];
+    public static byte[] escape(final byte[] bytes, final byte... appendAfterEscaping) {
+        final byte[] temp = new byte[2 * (bytes.length + ((appendAfterEscaping == null) ? 0 : appendAfterEscaping.length))];
```

Comment:
@gaffer01 Correct, this is a left over typo from when the appended bytes where being escaped, but this was incorrect behaviour and was append to the end without escaping. However as you spotted I didn't amend the initial array size.